src/mount/readdata.cc

```diff
@@ -410,6 +424,34 @@ void* read_data_delayed_ops(void *arg) {
 			}
 		}
 		lock.unlock();
+
+		ticksSinceLastUpdate++;
+		if (ticksSinceLastUpdate == 180) {
+			std::unique_lock lock(gReadCacheMemoryMutex);
+			if (gOriginalCacheExpirationTime_ms.load() == 0) {
+				gCacheExpirationTime_ms.store(0);
+			} else if (gCacheExpirationTime_ms.load() == 0) {
+				gCacheExpirationTime_ms.store(
+				    gOriginalCacheExpirationTime_ms.load());
+			} else if (timesRequestedMemory.load() > 0) {
+				double successRate =
+				    (double)successfulTimesRequestedMemory.load() /
+				    (double)timesRequestedMemory.load();
+				if (successRate < 0.3) {
+					gCacheExpirationTime_ms =
+					    std::max<uint32_t>(gCacheExpirationTime_ms / 3, 1);
+				} else if (successRate > 0.8) {
+					gCacheExpirationTime_ms = std::min<uint32_t>(
+					    gCacheExpirationTime_ms * 2,
+					    gOriginalCacheExpirationTime_ms.load());
+				}
+			}
+			timesRequestedMemory.store(0);
+			successfulTimesRequestedMemory.store(0);
+			lock.unlock();
+			ticksSinceLastUpdate = 0;
+		}
```

Comment:
Consider to move this part to an inline function. Also:
- Create constants for the magic numbers, e.g.: constexpr double kLowerSuccesRate = 0.3.
- Variable timesRequestedMemory is atomic, but is also used always protected by a lock on gReadCacheMemoryMutex. Atomic variables comes with a cost, if they are already protected, please consider using a regular variable.
- I think the same happens with successfulTimesRequestedMemory. Please check.

Reply:
Done